rollodex/src/App.js

```diff
@@ -1,26 +1,98 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {Component} from 'react';
+// import logo from './logo.svg';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isHidden: true, contacts: [],
+    }
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    fetch ('https://randomuser.me/api?results=25')
+    .then(response => response.json())
+    .then(parsedJSON => parsedJSON.results.map(contact => ({
+      name: `${contact.name.first} ${contact.name.last}`,
+      location: `${contact.location.city}, ${contact.location.state}, ${contact.location.country}`,
+      email: `${contact.email}`,
+      cell: `${contact.cell}`,
+      username: `${contact.login.username}`,
+      mediumpic: `${contact.picture.medium}`,
+      smallpic: `${contact.picture.thumbnail}`
+    })))
+    .then(contacts => this.setState({
+      contacts,
+      isHidden: true
```

Comment:
💯 Nice place to put this (`isHidden` I mean).

You wouldn't *necessarily* want to do this in an app in the "real world" since you're technically changing the data that's coming from the API but it all depends on how the project treats its APIs.  For instance if this roloodex allowed saving it might be problematic to try to save an `isHidden` attribute with the user's record so you'd have to make sure to exclude it.

Reply:
Derp, I misread this... I thought you were putting `isHidden` _into_ each contact's object 🤦‍♂